src/core/coap/coap_message.cpp

```diff
@@ -530,6 +564,26 @@ const otCoapOption *OptionIterator::GetNextOption(void)
     return rval;
 }
 
+otError OptionIterator::GetOptionValue(uint64_t &aValue) const
+{
+    otError error = OT_ERROR_NONE;
+    uint8_t value[sizeof(aValue)];
+    uint8_t pos = 0;
+
+    VerifyOrExit(mOption.mLength <= sizeof(aValue), error = OT_ERROR_NO_BUFS);
+    SuccessOrExit(GetOptionValue(value));
```

Comment:
How about 
```suggestion
    SuccessOrExit(error = GetOptionValue(value));
```

Reply:
Yep, will fix that.

Reply:
Fix is in cb96db62e9e5e2b92ea3da222b5c1a454af7facc.